File: StrormBookServer/src/main/java/com/api/stormbook/entity/Story.java
package com.api.stormbook.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
@Table(name = "stories")
public class Story {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String author;

    @Column(nullable = false)
    private String transGroup;

    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.PERSIST)
    @JoinTable(
            name = "story_category",
            joinColumns = @JoinColumn(name = "story_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private List<Category> categories;

    @Column(nullable = false)
    private String cover_image;

    @Column
    @Enumerated(EnumType.STRING)
    private Status status;

    @Column
    private int view_count;

    @Column
    private int like_count;

    @Column
    private double rating;

    @Column
    private int total_chapters;

    @Column(columnDefinition = "TEXT")
    private String description;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;


    public enum Status{
        OnGoing, Completed, Dropped
    }
}

File: StormBook/app/src/main/java/com/phongbaoto/stormbookv2/data/model/Story.kt
package com.phongbaoto.stormbookv2.data.model

data class Story(
    val id: Long,
    val author: String,
    val title: String,
    val trans_group: String,
    val cover_image: String,
    val created_at: String,
    val rating: Double,
    val status: STATUS,
    val updated_at: String,
    val view_count: Int,
    val like_count: Int,
    val description: String,
    val total_chapters: Int,
    val categoryId: List<Int>
)

enum class STATUS{
    Completed,
    Dropped,
    OnGoing
}